### PR TITLE
Purge expired Events

### PR DESCRIPTION
**Pull request by nefeline**
Ref. https://central.tri.be/issues/98826

### CODE REVIEW

File: src/Tribe/Event_Cleaner_Scheduler.php [hunk 1]
```diff
@@ -150,7 +150,14 @@ public function select_events_to_purge( $month ) {
 		INNER JOIN {$wpdb->postmeta} AS t2 ON t1.ID = t2.post_id
 		WHERE t1.post_type = %d
 			AND t2.meta_key = '_EventEndDate'
-			AND t2.meta_value <= DATE_SUB( CURDATE(), INTERVAL %d MONTH )";
+			AND t2.meta_value <= DATE_SUB( CURDATE(), INTERVAL %d MONTH )
+			AND t1.post_parent = 0
+			AND t1.ID NOT IN (
+				SELECT DISTINCT post_parent
+				FROM wp_posts
+				WHERE post_type= 'tribe_events'
```
Comment (bordoni) [inline src/Tribe/Event_Cleaner_Scheduler.php]: Shouldn't we use a variable for the Post Type here too?

File: src/Tribe/Event_Cleaner_Scheduler.php [hunk 2]
```diff
@@ -150,7 +150,14 @@ public function select_events_to_purge( $month ) {
 		INNER JOIN {$wpdb->postmeta} AS t2 ON t1.ID = t2.post_id
 		WHERE t1.post_type = %d
 			AND t2.meta_key = '_EventEndDate'
-			AND t2.meta_value <= DATE_SUB( CURDATE(), INTERVAL %d MONTH )";
+			AND t2.meta_value <= DATE_SUB( CURDATE(), INTERVAL %d MONTH )
+			AND t1.post_parent = 0
+			AND t1.ID NOT IN (
+				SELECT DISTINCT post_parent
```
Comment (bordoni) [inline src/Tribe/Event_Cleaner_Scheduler.php]: Maybe it would be worth saving this on a another SQL variable to make is more readable?
`AND t1.ID NOT IN ( $posts_with_parents_sql )`

File: src/Tribe/Event_Cleaner_Scheduler.php [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
 /**
- * Class Event_Scheduler
+ * Class Event_Cleaner_Scheduler
  *
  * Uses cron to move old events to trash and/or permanently delete them.
  *
  * @since TBD
  */
-class Tribe__Events__Event_Scheduler {
+class Tribe__Events__Event_Cleaner_Scheduler {
```
Comment (bordoni) [inline src/Tribe/Event_Cleaner_Scheduler.php]: Why the change in names?

Comment (barryhughes) [reply src/Tribe/Event_Cleaner_Scheduler.php]: I asked for this to make it less ambiguous (since the class exists specifically to deal with cleanup tasks whereas event scheduler is less explicit and could imply it handles publishing events on a schedule or something of that order).
